File: src/main/java/net/obesephoenix/koalakraftcrystals/events/KKEventHandler.java
package net.obesephoenix.koalakraftcrystals.events;

import org.bukkit.Bukkit;
import org.bukkit.event.Listener;
import org.bukkit.plugin.java.JavaPlugin;

public class KKEventHandler {

    public static void registerEvents(JavaPlugin plugin) {
        registerEvent(new onEntityDamage(), plugin);
        registerEvent(new onItemDespawn(), plugin);
        registerEvent(new onBlockPlace(), plugin);
        registerEvent(new onItemDrop(), plugin);
    }

    private static void registerEvent(Listener listener, JavaPlugin plugin) {
        plugin.getServer().getPluginManager().registerEvents(listener, plugin);
    }

}

File: src/main/java/net/obesephoenix/koalakraftcrystals/events/KKEventHandler.kt
package net.obesephoenix.koalakraftcrystals.events

import org.bukkit.plugin.java.JavaPlugin
import org.bukkit.event.Listener

object KKEventHandler {
    @JvmStatic
    fun registerEvents(plugin: JavaPlugin) {
        registerEvent(OnEntityDamage(), plugin)
        registerEvent(OnItemDespawn(), plugin)
        registerEvent(OnBlockPlace(), plugin)
        registerEvent(OnItemDrop(), plugin)
        registerEvent(OnMobEquip(), plugin)
    }

    private fun registerEvent(listener: Listener, plugin: JavaPlugin) {
        plugin.server.pluginManager.registerEvents(listener, plugin)
    }
}